aptos-move/framework/aptos-framework/sources/vesting.move

```diff
@@ -0,0 +1,451 @@
+        vesting_schedule: vector<FixedPoint32>,
+        withdrawal_address: address,
+        vesting_start_timestamp_secs: u64,
+        vesting_period_duration: u64,
+        operator: address,
+        voter: address,
+    ) acquires AdminNonce {
+        validate_vesting_account_inputs(
+            grant_amount,
+            &shareholders,
+            &shares,
+            &vesting_schedule,
+            withdrawal_address,
+            vesting_start_timestamp_secs,
+        );
+
+        // If this is the first time this admin account has created a vesting pool, initialize the nonce.
+        let admin_address = signer::address_of(admin);
+        if (!exists<AdminNonce>(admin_address)) {
+            move_to(admin, AdminNonce { vesting_pool_creation_nonce: 0 });
+        };
+
+        // Initialize the vesting pool in a new resource account. This allows the same admin to create multiple pools.
+        let (vesting_pool_account_signer, vesting_pool_account_signer_cap) = create_vesting_pool_account(admin);
+        stake::initialize_stake_owner(&vesting_pool_account_signer, 0, operator, voter);
+
+        // Extract owner_cap from the StakePool, so we have control over it in the delegations flow.
+        // This is stored as part of the delegation. Thus, the delegator would not have direct control over it without
+        // going through well-defined functions in this module.
+        let stake_pool_owner_cap = stake::extract_owner_cap(&vesting_pool_account_signer);
+
+        // Add the unvested amount to the stake pool.
+        let unvested_coins = coin::withdraw<AptosCoin>(admin, grant_amount);
+        stake::add_stake_with_cap(&stake_pool_owner_cap, unvested_coins);
+
+        move_to(&vesting_pool_account_signer, VestingPool {
+            state: VESTING_POOL_ACTIVE,
+            admin: admin_address,
+            grant_amount,
+            beneficiaries: simple_map::create<address, address>(),
+            shareholders,
+            shares: create_shares_map(&shareholders, &shares),
+            withdrawal_address,
+            stake_pool_owner_cap,
+            pool_address: signer::address_of(&vesting_pool_account_signer),
+            signer_cap: vesting_pool_account_signer_cap,
+
+            vesting_schedule,
+            vesting_start_timestamp_secs,
+            vesting_period_duration,
+            last_vesting_period: 0,
+        });
+    }
+
+    public entry fun batch_request_distribution(admin: &signer, pool_addresses: vector<address>) acquires VestingPool {
+        let len = vector::length(&pool_addresses);
+        let i = 0;
+        while (i < len) {
+            request_distribution(admin, *vector::borrow(&pool_addresses, i));
+            i = i + 1;
+        };
+    }
+
+    public entry fun request_distribution(admin: &signer, pool_address: address) acquires VestingPool {
+        let vesting_pool = borrow_global_mut<VestingPool>(pool_address);
+        verify_admin(admin, vesting_pool);
```

Comment:
why do we need admin?

Reply:
We don't. unlock_rewards, vest, and distribute now no longer require the admin